src/libsystemd/sd-netlink/generic-netlink.c

```diff
@@ -34,7 +34,9 @@ static int genl_message_new(sd_netlink *nl, sd_genl_family_t family, uint16_t nl
         size_t size;
         int r;
 
-        assert_return(nl->protocol == NETLINK_GENERIC, -EINVAL);
+        assert(nl);
+        assert(nl->protocol == NETLINK_GENERIC);
```

Comment:
was the change from assert_return to asset intentional?

Reply:
Yes, we should use `assert_return()` for exposed functions like `sd_netlink_xxx()` but not for internal ones.